internal/handlers/user.go

```diff
@@ -0,0 +1,429 @@
+	return token.SignedString([]byte(middleware.SecretKey))
+}
+
+func (uh *UserHandler) SaveOrder(w http.ResponseWriter, r *http.Request) {
+	username := r.Context().Value(middleware.UsernameKey).(string)
+
+	userID := uh.UserService.UserRepository.IsUserExists(username)
+
+	if userID < 0 {
+		http.Error(w, "пользователь не найден", http.StatusNotFound)
+		return
+	}
+
+	body, err := io.ReadAll(r.Body)
+
+	if err != nil {
+		http.Error(w, "Could not read body", http.StatusBadRequest)
+		return
+	}
+
+	defer r.Body.Close()
+
+	orderNumber := string(body)
+	isDigit := isDigits(orderNumber)
+
+	if !ValidateNumber(orderNumber) {
+		http.Error(w, "Неверный формат номера заказа", http.StatusUnprocessableEntity)
+		return
+	}
+
+	if !isDigit {
+		http.Error(w, "Неверный формат запроса", http.StatusBadRequest)
+		return
+	}
+
+	result, err := uh.OrderService.IsOrderExist(orderNumber, userID)
+
+	if err != nil {
+		http.Error(w, "Внутренняя ошибка сервера", http.StatusInternalServerError)
+		return
+	}
+
+	if result == 1 {
+		http.Error(w, "Номер заказа уже был загружен другим пользователем", http.StatusConflict)
+		return
+	} else if result == 2 {
+		http.Error(w, "Номер заказа уже был загружен этим пользователем", http.StatusOK)
+		return
+	}
+
+	err = uh.OrderService.SaveOrder(orderNumber, userID)
+
+	if err != nil {
+		http.Error(w, "Внутренняя ошибка сервера", http.StatusInternalServerError)
+		return
+	}
+
+	w.WriteHeader(http.StatusAccepted)
+
+	go func() {
+		tx, err := uh.OrderService.OrderRepository.DBStorage.Conn.BeginTx(
+			uh.OrderService.OrderRepository.DBStorage.Ctx, pgx.TxOptions{})
+
+		if err != nil {
+			http.Error(w, "Внутренняя ошибка сервера", http.StatusInternalServerError)
+			return
+		}
+
+		var registerResponse accrual.RegisterResponse
+		registerResponse, err = accrual.GetOrderInfo(uh.AccrualSystemAddress, orderNumber)
+
+		if err != nil {
+			http.Error(w, "Внутренняя ошибка сервера", http.StatusInternalServerError)
+			return
+		}
+
+		if registerResponse.Order != "" {
+			err = uh.OrderService.UpdateOrder(orderNumber, registerResponse.Accrual, registerResponse.Status)
+
+			if err != nil {
+				_ = tx.Rollback(uh.OrderService.OrderRepository.DBStorage.Ctx)
+				http.Error(w, "Внутренняя ошибка сервера", http.StatusInternalServerError)
+				return
+			}
+
+			err = uh.UserBalanceService.UpdateUserBalance(registerResponse.Accrual, userID)
+
+			if err != nil {
+				_ = tx.Rollback(uh.OrderService.OrderRepository.DBStorage.Ctx)
+				http.Error(w, "Внутренняя ошибка сервера", http.StatusInternalServerError)
+				return
+			}
+		}
+	}()
+}
+
+func isDigits(s string) bool {
+	re := regexp.MustCompile(`^\d+$`)
+	return re.MatchString(s)
+}
+
+func (uh *UserHandler) GetOrders(w http.ResponseWriter, r *http.Request) {
+	username := r.Context().Value(middleware.UsernameKey).(string)
+
+	userID := uh.UserService.UserRepository.IsUserExists(username)
+
+	if userID < 0 {
+		http.Error(w, "пользователь не найден", http.StatusNotFound)
```

Comment:
Мне кажется, что стоит форматировать сообщения об ошибках кронсистентно - либо с прописной, либо с заглавной буквы.
```suggestion
		http.Error(w, "Пользователь не найден", http.StatusNotFound)
```